### PR TITLE
Prof Edit Course Info Page

### PR DESCRIPTION
**Pull request by tiingweii-shii**
# Description
This is an implementation of the edit course info patch endpoint #811. 
On the admin panel, professors and course coordinators can be able to edit the course information (course display name, coordinator email, ical url, and crns). This PR only includes the backend implementation of this feature. 

Please include a summary of the change:
- added crns and icalURL field to the get course endpoint
- implemented the edit_course PATCH endpoint
- wrote tests for edit_course PATCH endpoint and updated a test snapshot for get course endpoint

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`

# How Has This Been Tested?

Please describe how you tested this PR (both manually and with tests)
Provide instructions so we can reproduce. 

- [x] Tested with Postman (example: PATCH localhost:3000/api/v1/courses/40/edit_course; body: { "courseId": 40, "name": "Potato", "coordinator_email": "potato@gmail.com", "icalURL": "potato.com", "crns": [12345, 56789] }
- [x] Added course integration tests

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/server/src/course/course.controller.ts [hunk 1]
```diff
@@ -160,7 +163,35 @@ export class CourseController {
       );
     }
 
-    return course;
+    const course_response: any = { ...course };
+    course_response.crns = null;
```
Comment (VKong6019) [inline packages/server/src/course/course.controller.ts:167]: nit: could probably have the `crns` field set to null inside the object above

File: packages/server/src/course/course.service.ts [hunk 2]
```diff
@@ -118,6 +120,86 @@ export class CourseService {
...
+          conflictCourse &&
+          conflictCourse.semesterId === course.semesterId
+        ) {
+          throw new BadRequestException(
+            ERROR_MESSAGES.courseController.crnAlreadyRegistered(crn, courseId),
+          );
+        }
+      }
+      if (
+        !courseCrnMap ||
```
Comment (dfarooq610) [inline packages/server/src/course/course.service.ts:160]: Do you need the logic for the second half of this if statement if you already check and throw the exception above?

Comment (tiingweii-shii) [reply packages/server/src/course/course.service.ts:160]: yes 🐻‍❄️

**Review by dfarooq610 — APPROVED** [review]
LGTM

File: packages/server/src/course/course.service.ts [hunk 3]
```diff
@@ -118,6 +120,86 @@ export class CourseService {
...
+        if (conflictCourse && conflictCourse.semesterId === course.semesterId) {
+          if (courseCrnMap.courseId !== courseId) {
+            throw new BadRequestException(
+              ERROR_MESSAGES.courseController.crnAlreadyRegistered(
+                crn,
+                courseId,
+              ),
+            );
+          } else {
+            courseCrnMapExists = true;
```
Comment (dfarooq610) [inline packages/server/src/course/course.service.ts:159]: nit: you can break out of the for loop here since it's true

**Review by dfarooq610 — CHANGES_REQUESTED** [review]
smol nits

**Review by dfarooq610 — APPROVED** [review]
Looks Good!

**Review by sumitde22 — APPROVED** [review]
Looks good to me

**Pull request merged**